test/fixtures/for-in/array-scope-register-declaration/expected.js

```diff
@@ -0,0 +1,15 @@
+function f() {
+  const _array = g();
+
+  for (let _i = 0, _len = _array.length; _i < _len; _i++) {
+    const k = _array[_i];
+    k;
+  }
+
+  const _array2 = g();
+
+  for (let _i2 = 0, _len2 = _array2.length; _i2 < _len2; _i2++) {
```

Comment:
This might be a `registerBinding` thing, or I may be calling uid on the wrong scope. Another possibility is that Babel is anticipating compiling down to ES5, in which case for loops don't get their own scope...

At any rate, can we consider this non blocking for the moment?